In [1]:
import pandas as pd

In [2]:
dataset=pd.read_csv("Placement_Pre-processed_Dataset.csv")

In [3]:
dataset

sl_no  ssc_p  hsc_p  degree_p  etest_p  mba_p    salary gender    ssc_b  \
0      1.0  67.00  91.00     58.00     55.0  58.80  270000.0      M   Others   
1      2.0  79.33  78.33     77.48     86.5  66.28  200000.0      M  Central   
2      3.0  65.00  68.00     64.00     75.0  57.80  250000.0      M  Central   
3      4.0  56.00  52.00     52.00     66.0  59.43       0.0      M  Central   
4      5.0  85.80  73.60     73.30     96.8  55.50  425000.0      M  Central   
..     ...    ...    ...       ...      ...    ...       ...    ...      ...   
210  211.0  80.60  82.00     77.60     91.0  74.49  400000.0      M   Others   
211  212.0  58.00  60.00     72.00     74.0  53.62  275000.0      M   Others   
212  213.0  67.00  67.00     73.00     59.0  69.72  295000.0      M   Others   
213  214.0  74.00  66.00     58.00     70.0  60.23  204000.0      F   Others   
214  215.0  62.00  58.00     53.00     89.0  60.22       0.0      M  Central   

       hsc_b     hsc_s   degree_t workex specialisation      status  
0     Others  Commerce   Sci&Tech     No         Mkt&HR      Placed  
1     Others   Science   Sci&Tech    Yes        Mkt&Fin      Placed  
2    Central      Arts  Comm&Mgmt     No        Mkt&Fin      Placed  
3    Central   Science   Sci&Tech     No         Mkt&HR  Not Placed  
4    Central  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
..       ...       ...        ...    ...            ...         ...  
210   Others  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
211   Others   Science   Sci&Tech     No        Mkt&Fin      Placed  
212   Others  Commerce  Comm&Mgmt    Yes        Mkt&Fin      Placed  
213   Others  Commerce  Comm&Mgmt     No         Mkt&HR      Placed  
214   Others   Science  Comm&Mgmt     No         Mkt&HR  Not Placed  

[215 rows x 15 columns]

In [4]:
dataset.isnull().sum()

sl_no             0
ssc_p             0
hsc_p             0
degree_p          0
etest_p           0
mba_p             0
salary            0
gender            0
ssc_b             0
hsc_b             0
hsc_s             0
degree_t          0
workex            0
specialisation    0
status            0
dtype: int64

In [5]:
def QuanQual():
    Quan=[]
    Qual=[]
    for columnName in dataset.columns:
        print(columnName)
        if(dataset[columnName].dtypes=="O"):
            #print("Qual")
            Qual.append(columnName)
        else:
            #print("Quan")
            Quan.append(columnName)
    return Quan,Qual

In [6]:
Quan,Qual=QuanQual()

sl_no
ssc_p
hsc_p
degree_p
etest_p
mba_p
salary
gender
ssc_b
hsc_b
hsc_s
degree_t
workex
specialisation
status


In [7]:
dir(dataset)

['T',
 '_AXIS_LEN',
 '_AXIS_ORDERS',
 '_AXIS_TO_AXIS_NUMBER',
 '_HANDLED_TYPES',
 '__abs__',
 '__add__',
 '__and__',
 '__annotations__',
 '__array__',
 '__array_priority__',
 '__array_ufunc__',
 '__arrow_c_stream__',
 '__bool__',
 '__class__',
 '__contains__',
 '__copy__',
 '__dataframe__',
 '__dataframe_consortium_standard__',
 '__deepcopy__',
 '__delattr__',
 '__delitem__',
 '__dict__',
 '__dir__',
 '__divmod__',
 '__doc__',
 '__eq__',
 '__finalize__',
 '__floordiv__',
 '__format__',
 '__ge__',
 '__getattr__',
 '__getattribute__',
 '__getitem__',
 '__getstate__',
 '__gt__',
 '__hash__',
 '__iadd__',
 '__iand__',
 '__ifloordiv__',
 '__imod__',
 '__imul__',
 '__init__',
 '__init_subclass__',
 '__invert__',
 '__ior__',
 '__ipow__',
 '__isub__',
 '__iter__',
 '__itruediv__',
 '__ixor__',
 '__le__',
 '__len__',
 '__lt__',
 '__matmul__',
 '__mod__',
 '__module__',
 '__mul__',
 '__ne__',
 '__neg__',
 '__new__',
 '__nonzero__',
 '__or__',
 '__pandas_priority__',
 '__pos__',
 '__pow__',
 '__r

In [8]:
dataset[Quan].cov()

sl_no          ssc_p          hsc_p       degree_p  \
sl_no      3870.000000     -52.641355     -58.106028     -40.413645   
ssc_p       -52.641355     117.228377      60.348373      42.897137   
hsc_p       -58.106028      60.348373     118.755706      34.819820   
degree_p    -40.413645      42.897137      34.819820      54.151103   
etest_p      52.556168      37.659225      35.461678      21.929469   
mba_p         8.102336      24.535952      22.555846      17.272020   
salary    24490.654206  901754.893936  763359.777657  465131.504238   

                etest_p          mba_p        salary  
sl_no         52.556168       8.102336  2.449065e+04  
ssc_p         37.659225      24.535952  9.017549e+05  
hsc_p         35.461678      22.555846  7.633598e+05  
degree_p      21.929469      17.272020  4.651315e+05  
etest_p      176.251018      16.886973  3.842344e+05  
mba_p         16.886973      34.028376  1.262455e+05  
salary    384234.419257  126245.485547  2.395714e+10

#The positive covariance of 16.89 between etest_p (employability test score) and mba_p (MBA percentage) suggests that students who score higher on the employability test also tend to score higher in their MBA. This indicates that there is a general trend where good performance in one area is associated with good performance in the other.
#For students, this positive covariance means that there is a tendency for those who perform well on their employability tests to also perform well in their degree courses. 
When you do better on your degree exams, you are also likely to score higher on employability tests, and vice versa.

In [10]:
dataset[Quan].corr()

sl_no     ssc_p     hsc_p  degree_p   etest_p     mba_p    salary
sl_no     1.000000 -0.078155 -0.085711 -0.088281  0.063636  0.022327  0.002543
ssc_p    -0.078155  1.000000  0.511472  0.538404  0.261993  0.388478  0.538090
hsc_p    -0.085711  0.511472  1.000000  0.434206  0.245113  0.354823  0.452569
degree_p -0.088281  0.538404  0.434206  1.000000  0.224470  0.402364  0.408371
etest_p   0.063636  0.261993  0.245113  0.224470  1.000000  0.218055  0.186988
mba_p     0.022327  0.388478  0.354823  0.402364  0.218055  1.000000  0.139823
salary    0.002543  0.538090  0.452569  0.408371  0.186988  0.139823  1.000000

#Correlation measures how strongly two variables are related to each other. A positive correlation means that when one variable increases, the other tends to increase as well.
For mba_p (MBA percentage) and salary, the correlation is 0.14. This is a positive correlation, although it's quite small. This means that generally, as MBA percentages increase, salaries also tend to increase slightly. However, the correlation is not very strong, indicating that MBA scores do not have a very large impact on salary.
 while there is a slight tendency for those with higher MBA percentages to have higher salaries, the connection isn't very strong. Other factors might have a larger influence on salary.

# Reject Null hypothesis p<0.05
# Accept Null Hypothesi and Reject Alternate hypothsis

# ANAVO : Analysis of Variance

In [14]:
#ONE WAY
import scipy.stats as stats
stats.f_oneway(dataset['ssc_p'],dataset['hsc_p'],dataset['degree_p'])

F_onewayResult(statistic=0.6719700864663097, pvalue=0.5110602818995302)

In [15]:
##2. Two Way Classification: Two Independent Variable
import statsmodels.api as sm
from statsmodels.formula.api import ols
anova_two=ols('sl_no ~ ssc_p + hsc_p + degree_p + ssc_p*hsc_p*degree_p',data=dataset).fit()
Two=sm.stats.anova_lm(anova_two,type=2)
Two

df         sum_sq      mean_sq         F    PR(>F)
ssc_p                   1.0    5058.655954  5058.655954  1.299536  0.255615
hsc_p                   1.0    2346.272810  2346.272810  0.602742  0.438420
degree_p                1.0    1620.170837  1620.170837  0.416211  0.519548
ssc_p:hsc_p             1.0    9473.060349  9473.060349  2.433567  0.120290
ssc_p:degree_p          1.0    3109.834156  3109.834156  0.798896  0.372462
hsc_p:degree_p          1.0     315.719461   315.719461  0.081106  0.776089
ssc_p:hsc_p:degree_p    1.0     474.744742   474.744742  0.121959  0.727275
Residual              207.0  805781.541691  3892.664453       NaN       NaN

In [ ]:
#### Reject Null hypothesis p<0.05
# Accept Null Hypothesi and Reject Alternate hypothsis
#h0=There is no similarity interaction effect b/t ssc_p*hsc_p*degree_p
#h1=There is  similarity interaction effect b/t ssc_p*hsc_p*degree_p
